src/test/java/core/basesyntax/service/impl/ShopServiceImplTest.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.operation.BalanceOperation;
+import core.basesyntax.strategy.operation.OperationHandler;
+import core.basesyntax.strategy.operation.PurchaseOperation;
+import core.basesyntax.strategy.operation.ReturnOperation;
+import core.basesyntax.strategy.operation.SupplyOperation;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Testing ShopServiceImpl")
+class ShopServiceImplTest {
```

Comment:
probably you need to add additional tests here

Reply:
Ok

Reply:
Ok